.github/actions/secure-upload-folder/action.yml

```
@@ -0,0 +1,32 @@
+name: "Secure folder upload"
+description: "Upload a folder as a tarball and outputs its SHA256"
+inputs:
+  name:
+    description: "Artifact name."
+    default: "artifact"
+    required: false
+  path:
+    description: 'Path to the folder'
+    required: true
+outputs:
+  sha256:
+    description: "SHA256 of the uploaded tarball."
+    value: "${{ steps.upload.outputs.sha256 }}"
+
+runs:
+  using: "composite"
+  steps:
+    - name: Create tarball
+      env:
+        PATH: ${{ inputs.path }}
+      run: |
+        tar -cvf "$PATH" "$PATH.tar"
+
+    - name: Upload the artifact
+      id: upload
+      uses: slsa-framework/slsa-github-generator/.github/actions/secure-upload-artifact@main
+      with:
+        name: "${{ inputs.name }}"
+        path: "${{ inputs.path }}.tar"
+        if-no-files-found: error
+        retention-days: 5
```

More extensive suggestions.

- Compress the tar file
- `tar` will overwrite the file silently. Write it to a temp directory.
- Don't overwrite `$PATH`
```suggestion
      id: create
      env:
        DIRPATH: ${{ inputs.path }}
      run: |
        # check if a directory
        if [[ ! -d "${DIRPATH}" ]]; then
            >&2 echo "Error: ${DIRPATH} is not a directory."
            exit 1
        fi
        tempdir=$(mktemp -d)
        foldername=$(basename "${DIRPATH}")
        tarball_path="${tempdir}/${foldername}.tgz"
        tar -czvf "${tarball_path}" "${DIRPATH}"
        echo "tarball-path=${tarball_path}" >> "$GITHUB_OUTPUT"

    - name: Upload the artifact
      id: upload
      uses: slsa-framework/slsa-github-generator/.github/actions/secure-upload-artifact@main
      with:
        name: "${{ inputs.name }}"
        path: "${{ steps.create.outputs.tarball-path }}"
        if-no-files-found: error
        retention-days: 5
```